spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -534,6 +535,9 @@ public void run() {
 			if (this.genericListener instanceof ConsumerSeekAware) {
 				((ConsumerSeekAware) this.genericListener).registerSeekCallback(this);
 			}
+			if (this.transactionManager != null) {
+				ProducerFactoryUtils.setContainerGroupId(this.consumerGroupId);
```

Comment:
OK. Everything is good. 
I get back my suggestion about `Acknowledge`.
Indeed we need that `Producer` resource for transactions...

Only this issue about container - consumer inconsistencty.